app/src/main/java/com/amaze/filemanager/database/typeconverters/OpenModeTypeConverter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2014-2020 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
+ * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
+ *
+ * This file is part of Amaze File Manager.
+ *
+ * Amaze File Manager is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.database.typeconverters;
+
+import com.amaze.filemanager.utils.OpenMode;
+
+import androidx.room.TypeConverter;
+
+/** {@link TypeConverter} for {@link OpenMode} objects to database columns. */
+public class OpenModeTypeConverter {
+
+  @TypeConverter
+  public static int fromOpenMode(OpenMode from) {
```

Comment:
Is it okay to have it to convert to int? Since older tables had `OpenMode` as column data type, I think that would've been saving it's `toString()` value.

Reply:
It only deals with converting the `enum` to database value and vice versa. And as far as I can tell from current state of `cloud` table `service` column is still of `INTEGER` type.

Reply:
Also, per database schema, `cloud.service` had been INTEGER from the beginning. Specifying `OpenModeTypeConverter` at CloudEntry

https://github.com/TeamAmaze/AmazeFileManager/blob/fd0820c88f201d4591fff0ccc5c4d72c805c7bc8/app/src/main/java/com/amaze/filemanager/database/models/explorer/CloudEntry.java#L42-L44

Room will convert the enum to integer (i.e. the `OpenMode`'s enum numeric value) and store to database; and on the other hand, convert the numeric value back to `OpenMode` when fetching from database.

Same serialization/deserialization is similar those we can see in other ORM libraries in the Java world too.

So far I didn't see any other tables that would store `OpenMode` as string - and it's possible to specify another `TypeConverter` implementation in that case.

Hope this can explain things clear enough.

Reply:
Ahh, you're right.. it was saving `openMode#ordinal` earlier as well. Sorry my bad.